SDL/Quintuplet.cu

```diff
@@ -0,0 +1,646 @@
+
+#ifdef CACHE_ALLOC
+    cudaStream_t stream = 0;
+    quintupletsInGPU.tripletIndices = (unsigned int*)cms::cuda::allocate_managed(nMemoryLocations * 2 * sizeof(unsigned int));
+    quintupletsInGPU.lowerModuleIndices = (unsigned int*)cms::cuda::allocate_managed(nMemoryLocations * 5 * sizeof(unsigned int));
+    quintupletsInGPU.nQuintuplets = (unsigned int*)cms::cuda::allocate_managed(nLowerModules * sizeof(unsigned int));
+#else
+    cudaMallocManaged(&quintupletsInGPU.tripletIndices, 2 * nMemoryLocations * sizeof(unsigned int));
+    cudaMallocManaged(&quintupletsInGPU.lowerModuleIndices, 5 * nMemoryLocations * sizeof(unsigned int)); 
+
+    cudaMallocManaged(&quintupletsInGPU.nQuintuplets, nLowerModules * sizeof(unsigned int));
+
+#ifdef CUT_VALUE_DEBUG
+    cudaMallocManaged(&quintupletsInGPU.innerRadius, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.innerRadiusMin, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.innerRadiusMax, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.bridgeRadius, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.bridgeRadiusMin, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.bridgeRadiusMax, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.outerRadius, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.outerRadiusMin, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.outerRadiusMax, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.innerRadiusMin2S, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.innerRadiusMax2S, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.bridgeRadiusMin2S, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.bridgeRadiusMax2S, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.outerRadiusMin2S, maxQuintuplets * nLowerModules * sizeof(float));
+    cudaMallocManaged(&quintupletsInGPU.outerRadiusMax2S, maxQuintuplets * nLowerModules * sizeof(float));
+#endif
+#endif
+
+#pragma omp parallel for
+    for(size_t i = 0; i<nLowerModules;i++)
+    {
+        quintupletsInGPU.nQuintuplets[i] = 0;
+    }
+
+}
+
+void SDL::createQuintupletsInExplicitMemory(struct SDL::quintuplets& quintupletsInGPU, const unsigned int& maxQuintuplets, const unsigned int& nLowerModules, const unsigned int& nEligibleModules)
+{
+    unsigned int nMemoryLocations = nEligibleModules * maxQuintuplets;
+#ifdef CACHE_ALLOC
+    cudaStream_t stream = 0;
+    int dev;
+    cudaGetDevice(&dev);
+    cudaMalloc(&quintupletsInGPU.tripletIndices, 2 * nMemoryLocations * sizeof(unsigned int));
```

Comment:
Shouldn't these also use cache allocation?

Reply:
I copied this from Track candidates (https://github.com/SegmentLinking/TrackLooper/blame/master/SDL/TrackCandidate.cu#L93-L95)

here for some reason some objects are not cache allocated

Reply:
Those objects exceed the maximum size for cache allocation so they are excluded. The quintuple objects can be allocated using caching just fine